### PR TITLE
Adds Chainsaw

### PR DESCRIPTION
**Pull request by MisterLayne**
Oh shit, looks like another weapon to be passed off as a tool.

- [x] Replace placeholder sprites with legit ones.
- [x] Add chainsaw sounds.

nice_chainsaw

Source for sounds (edited them):
    https://myaudiograb.com/hE6P0Lzk

### CODE REVIEW

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+obj/item/weapon/material/twohanded/chainsaw/afterattack(atom/A as mob|obj|turf|area, mob/user as mob, proximity)
+	if(!proximity) return
+	..()
+	if(A && wielded && on)
+		if(istype(A,/obj/structure/window))
+			var/obj/structure/window/W = A
+			W.shatter()
+		else if(istype(A,/obj/structure/grille))
+			qdel(A)
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: No rods at all?

Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Wasn't sure how to make it not no.. Not.. Uh..
How to make rods be there. 
How to PROPERLY break the grille.

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+obj/item/weapon/material/twohanded/chainsaw/proc/get_fuel()
+	reagents.get_reagent_amount("fuel")
+
+obj/item/weapon/material/twohanded/chainsaw/examine(mob/user)
+	if(..(user,0))
+		if(max_fuel)
+			to_chat(usr, "<span class = 'notice'>The [src] feels like it contains roughtly [get_fuel()] units of fuel left.</span>")
+
+obj/item/weapon/material/twohanded/chainsaw/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is laying down and pulling the chainsaw into him, it looks like \he's trying to commit suicide!</span>")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: \him

Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Pushed just as you reviewed B)

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	if(max_fuel <= 0)
+		if(do_after(usr, 15))
+			to_chat(usr, "\The [src] won't start!")
+		else
+			to_chat(usr, "You fumble with the string.")
+	else if(!wielded)
+		to_chat(usr, "You need both hands to start the chainsaw!")
+	else
+		if(do_after(usr, 15))
+			to_chat(usr, "\The [src] starts up with a loud noise!")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: Should probably be more than just to usr, no?

Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Probably.

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 1 above)
Comment (Belsima) [reply code/game/objects/items/weapons/material/chainsaw.dm]: I think just spawning a small bit of rods works- I'm fairly sure that's all grilles do.

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	var/datum/reagents/R = new/datum/reagents(max_fuel)
+	reagents = R
+	R.my_atom = src
+	R.add_reagent("fuel", max_fuel)
+	..()
+
+obj/item/weapon/material/twohanded/chainsaw/proc/turnOn()
+	if(on) return
+
+	to_chat(usr, "You start pulling the string on \the [src].")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: Also probably more than usr. Visible message seems appropriate.

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	w_class = ITEMSIZE_LARGE
+	slot_flags = SLOT_BACK
+	unwielded_force_divisor = 0.25
+	force_divisor = 0.5
+	dulled_divisor = 0.6
+	sharp = 1
+	edge = 1
+	w_class = ITEMSIZE_LARGE
+	slot_flags = SLOT_BACK
+	force_wielded = 35
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: Is it intended to be so high up here?

**Comment by Belsima** [comment]
How does it work on vines and blobs?

**Comment by Belsima** [comment]
Also should maybe be ghetto surgery circular saw standin

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 5 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Force_wielded?

Comment (Belsima) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Shouldn't force only be Super High when it's on, or am I misinterpreting this?

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	if(max_fuel <= 0)
+		if(do_after(usr, 15))
+			to_chat(usr, "\The [src] won't start!")
+		else
+			to_chat(usr, "You fumble with the string.")
+	else if(!wielded)
+		to_chat(usr, "You need both hands to start the chainsaw!")
+	else
+		if(do_after(usr, 15))
+			visible_message("You start \the [src] up with a loud grinding!", "[usr] starts \the [src] up with a loud grinding!")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: I think the 'you' comes second, but that may be old code.

**Comment by Atermonera** [comment]
Well according to travis it took at least 10 minutes to load the map files. I'd hazard a guess that it's to do with #4307 because this is the first time I've ever seen this particular error in travis, but it's also so far unique among the pulls after that was merged

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	var/datum/reagents/R = new/datum/reagents(max_fuel)
+	reagents = R
+	R.my_atom = src
+	R.add_reagent("fuel", max_fuel)
+	..()
+
+obj/item/weapon/material/twohanded/chainsaw/proc/turnOn()
+	if(on) return
+
+	visible_message("You start pulling the string on \the [src].", "[usr] starting pulling the string on the [src].")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: `Urist McAssistant starting pulling the string on the chainsaw.`

Maybe `[usr] starts pulling the string on the [src].`

File: code/game/objects/items/weapons/material/chainsaw.dm [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+obj/item/weapon/material/twohanded/chainsaw/proc/get_fuel()
+	reagents.get_reagent_amount("fuel")
+
+obj/item/weapon/material/twohanded/chainsaw/examine(mob/user)
+	if(..(user,0))
+		if(max_fuel)
+			to_chat(usr, "<span class = 'notice'>The [src] feels like it contains roughtly [get_fuel()] units of fuel left.</span>")
+
+obj/item/weapon/material/twohanded/chainsaw/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is laying down and pulling the chainsaw into \him, it looks like \he's trying to commit suicide!</span>")
```
Comment (Belsima) [inline code/game/objects/items/weapons/material/chainsaw.dm]: lying down

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 6 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Checked other examples, they had this format, but once again, may be old code. I'll look into it.

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 7 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: Grammatical error. Was tired when made fixes. Will fix when get home.

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 8 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: My English teacher would be disappointed in me.

File: code/game/objects/items/weapons/material/chainsaw.dm (hunk 5 above)
Comment (MisterLayne) [reply code/game/objects/items/weapons/material/chainsaw.dm]: True that. Will handle.

**Comment by Anewbe** [comment]
What is this supposed to do, exactly? I'm reading it as a weaker fireaxe, basically?

**Comment by Spacemanspark** [comment]
Might want to look at the sprites on the chainsaws that /tg/ servers currently have.

**Comment by Belsima** [comment]
@Anewbe Weaker if off, stronger if on.

**Comment by Anewbe** [comment]
Ah, there it is.

With those numbers it's **still** just a weaker fireaxe, though, if I'm remembering my numbers correctly.

**Comment by MisterLayne** [comment]
Could implement a use for trees on Southern Cross.. Also, really badass. ALSO- I could always make it STRONGER than the fireaxe, and balance it by making it use less gas?

**Comment by Anewbe** [comment]
You can already cut down trees with any sharp edged item.
And I guess you could. I'm not saying I don't want this, I'm just not sure what niche it's trying to fill.

**Comment by MisterLayne** [comment]
Heyyy, who doesn't like cliche serial killers?
I mean, sharp items can, but can they do it FASTER than the chainsaw could?

**Comment by Anewbe** [comment]
Mechanically, yes.

**Comment by mistyLuminescence** [comment]
In my opinion, something doesn't really need to fill a missing niche to be worth adding - if it's cool, and if it's not buggy, and if it wouldn't break the game, and if there's not a big outcry about it, why not? 

Chainsaws are objectively cool.
This code seems not too buggy.
I doubt it'd break the game if fireaxes don't.
And there's not a big outcry about it.

Agreed on taking the tg sprites for chainsaws.

If you add in a check to cut down a tree after a time delay, it'd fill the lumberjack niche as well.

**Comment by Belsima** [comment]
@Anewbe Base force for a fireaxe is 30, base for this when on is 40.

**Comment by Anewbe** [comment]
Noooo? It's 42 per hit on a fireaxe.

**Comment by Belsima** [comment]
Ah, I just got mine from glancing at the base in the code.

**Comment by Anewbe** [comment]
@MisterLayne Is this good to go?

**Comment by MisterLayne** [comment]
As long as you think it's balanced, should be good @Anewbe .

**Pull request merged**